### PR TITLE
Disable buttons after first click [MAILPOET-2273]

### PR DESCRIPTION
**Pull request by amine-mp**
[MAILPOET-2273]

[MAILPOET-2273]: https://mailpoet.atlassian.net/browse/MAILPOET-2273

### CODE REVIEW

File: assets/js/src/newsletters/types.jsx [hunk 1]
```diff
@@ -23,7 +30,7 @@ class NewsletterTypes extends React.Component {
     }
   };
 
-  getAutomaticEmails = () => {
+  getAutomaticEmails() {
```
Comment (JanJakes) [inline assets/js/src/newsletters/types.jsx]: Why did you change the type of the function? I see you used `.bind(this)` below so it's OK I guess but is there any reason? Now some functions in this class are arrow functions, some are not.

File: assets/js/src/newsletters/types.jsx [hunk 2]
```diff
@@ -73,98 +81,96 @@ class NewsletterTypes extends React.Component {
...
+                event.preventDefault();
+                createWelcomeNewsletter();
+              }
+            }}
+            role="button"
+            tabIndex={0}
+          >
+            {MailPoet.I18n.t('setUp')}
+          </a>
+        ),
```
Comment (JanJakes) [inline assets/js/src/newsletters/types.jsx]: I see you unwrapped the closure and then added the `disabled` attribute. It's all in one commit so I don't know why. Then I have to ask - did you unwrap it just because it's nicer or because it's needed?

File: assets/js/src/newsletters/types.jsx (hunk 1 above)
Comment (amine-mp) [reply assets/js/src/newsletters/types.jsx]: Normally all methods should be functions in order to access `this` properly. But ESLint complains about using a `function` and not using `this` inside it. So we ended up converting those to `arrow functions`. Now I changed this to be a `function` because I needed to use `this` inside.

File: assets/js/src/newsletters/types.jsx (hunk 2 above)
Comment (amine-mp) [reply assets/js/src/newsletters/types.jsx]: Because it's needed. if I keep the closure/anonymous function, then I can't access the class instance using `this`.

File: assets/js/src/newsletters/types.jsx (hunk 1 above)
Comment (JanJakes) [reply assets/js/src/newsletters/types.jsx]: I see. However:
1. Arrow functions do allow you to access the correct `this` without the need to bind it.
2. It may be indeed better to use bound functions in some cases (see https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1) but in such case, the `bind()` belongs to the constructor, not as a side-effect of `render` method (https://github.com/mailpoet/mailpoet/pull/2361/files#diff-e7956c90afbfb37c0f8a5bca3e5ebfa3R95).

**Comment by amine-mp** [comment]
You are right @JanJakes, there is no need to change methods from arrow functions because I removed closures on action render, which were causing the actual issue.
